### DISCUSSION TITLE
Overriding the macOS built-in Python installation

### DISCUSSION THREAD

**Discussion by carlescufi** [Homebrew/discussions#4893] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.19
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7f745d04d90eaec76c8d5830d6ce6a0d7db888b3
Last commit: 3 days ago
Core tap JSON: 09 Nov 17:32 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_blizzard_avalanche
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: 14.3.1.0.1.1683849156
Xcode: 14.3.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Description of issue

Is it still necessary to add the line below in your `.zshrc` or `.zprofile` in order for Homebrew's Python to become the default and override the one that comes with the OS in `/usr/bin`?

`export PATH="/opt/homebrew/opt/python/libexec/bin:$PATH"`

**Comment by Bo98** [comment, marked as answer]
Only if you need `python` rather than just `python3` (similarly with `pip` vs `pip3` etc)

**Reply by carlescufi** [reply]
Thanks, I had `python` aliased to `python3` so I wasn't seeing that. However, now that I've removed the alias I actually can't find any `python` executable or symlink in `/usr/bin`, so without adding `/opt/homebrew/opt/python/libexec/bin` to the `PATH` I simply get `command not found` when executing `$ python`. Is this expected, does Apple only provide a `python3` executable but no `python`?

**Reply by Bo98** [reply]
Apple removed their `python` (2.7) in macOS 12.3.

**Reply by andrewcrook** [reply]
Didn't they remove Python3 as well? I  thought on a fresh install of latest macOS running python3 for the first time downloads python. can anyone verify this? My understanding from macOS dev notes is that Apple is going to remove most dependences on various interpreters and not to include them in the future. Software developers are recommended to provide their own installation of interpreters required for their apps. For users and developers they will have to install them manually. When I say "remove most “ I am thinking python and ruby the future systems will have shells and I presume perl because of its heavy unix usage but I am not even 100% sure of perl being included. Personally I use asdf to manage python and ruby etc

**Reply by Bo98** [reply]
Sort of. There is no python3 shipped with the OS, but there is one shipped with Xcode/CLT.

Ruby is unlikely to ever be updated again. It'll take a while to remove because of how many things depend on it (including us, but not for much longer).

Perl is an interesting one as they still update it. Perl 5.34 shipped with Sonoma.

**Reply by andrewcrook** [reply]
thanks 

irc  Xcode/CLT uses python compiled as a framework rather than your standard version iirc so does homebrew on macOS.
Ruby agreed,  Apple used to use Ruby quite a bit for scripts under the hood.
I think Perl will eventually go there is no Perl 6 now it's been replaced by Raku which doesn’t seem to gain as much interest.
I thought a lot more people started using python over perl these days.
Swift script release soon maybe ??
